api_trial_system.yaml

```diff
@@ -0,0 +1,163 @@
+swagger: '2.0'
+info:
+  version: 13.34.3
+  title: Trial System API
+host: localhost
+basePath: /trials/api/v1
+schemes:
+  - https
+security:
+  - Bearer: []
+securityDefinitions:
+  Bearer:
+    type: apiKey
+    name: Authorization
+    in: header
+paths:
+  /subscriptions:
+    get:
+      operationId: getSubscription
+      summary: Get subscription detail
+      description: >-
+        Returns the details about the subscription of the authenticated user for
+        the specified trial.
+      parameters:
+        - $ref: '#/parameters/queryTrialId'
+      responses:
+        '200':
+          $ref: '#/responses/200Subscription'
+        '401':
+          $ref: '#/responses/401Unauthorized'
+        '429':
+          $ref: '#/responses/429TooManyRequest'
+        '500':
+          $ref: '#/responses/500Error'
+    post:
+      operationId: createSubscription
+      summary: Create a subscription
+      description: Subscribe the authenticated user to the given trial.
+      parameters:
+        - $ref: '#/parameters/bodyCreateSubscription'
+      responses:
+        '200':
+          $ref: '#/responses/200Subscription'
+        '400':
+          $ref: '#/responses/400Error'
+        '401':
+          $ref: '#/responses/401Unauthorized'
+        '429':
+          $ref: '#/responses/429TooManyRequest'
+        '500':
+          $ref: '#/responses/500Error'
+parameters:
+  queryTrialId:
+    name: trialId
+    in: query
```

Comment:
why query param instead of path param?

Reply:
I initially chose to use it as a query parameter because it didn't seem to me that the `trialId` was the identifier of the subscription. However, upon further consideration, I realized that the `trialId` uniquely identifies the subscription for the user. Therefore, it would be more appropriate to use it as a path parameter. I'll make that update! 🙂

Reply:
Done ✅